frontend/packages/dev-console/src/components/topology/topology-utils.ts

```diff
@@ -35,8 +35,10 @@ export const getServiceBindingStatus = ({ FLAGS }: RootState): boolean =>
 export const getCheURL = (consoleLinks: K8sResourceKind[]) =>
   _.get(_.find(consoleLinks, ['metadata.name', 'che']), 'spec.href', '');
 
-export const getEditURL = (gitURL: string, cheURL: string) => {
-  return gitURL && cheURL ? `${cheURL}/f?url=${gitURL}&policies.create=peruser` : gitURL;
+export const getEditURL = (gitURL: string, gitBranch: string, cheURL: string) => {
+  return gitURL && cheURL
+    ? `${cheURL}/f?url=${gitURL}/tree/${gitBranch}&policies.create=peruser`
+    : gitURL;
```

Comment:
```suggestion
export const getEditURL = (gitURL: string, gitBranch: string, cheURL: string) => {
  const gitBranchRef = gitBranch ? `/tree/${gitBranch}` : '';
  return gitURL && cheURL
    ? `${cheURL}/f?url=${gitURL}${gitBranchRef}&policies.create=peruser`
    : gitURL;
};
```
To handle the case when we we do not have vcsRef annotation on the resource

Reply:
Updated.